### PR TITLE
AB#34697: Custom Page Creator

### PR DESCRIPTION
**Pull request by bradthompson995**
## Overview

This implements the Custom Page Functionality - including both frontend and backend.

- Creating & Updating Custom Pages
- Custom Page ADAC Index
- Ability to access custom pages

Custom pages follow the following route schema:
/pages/**slug** (slug is the unique identifier used for each custom pages to access it.)

### Azure Boards
- [AB#34697](https://dev.azure.com/UniversityOfNottingham/bd2de5ba-2a41-4bdc-a444-61e6b706132a/_workitems/edit/34697)
- [AB#34240](https://dev.azure.com/UniversityOfNottingham/bd2de5ba-2a41-4bdc-a444-61e6b706132a/_workitems/edit/34240)
- [AB#34713](https://dev.azure.com/UniversityOfNottingham/bd2de5ba-2a41-4bdc-a444-61e6b706132a/_workitems/edit/34713)

### CODE REVIEW

File: src/Directory/Biobanks.Web/packages.config [hunk 1]
```diff
@@ -57,10 +57,10 @@
   <package id="Microsoft.Owin.Security.Cookies" version="4.1.0" targetFramework="net48" />
   <package id="Microsoft.Owin.Security.OAuth" version="4.1.0" targetFramework="net48" />
   <package id="Microsoft.Web.Infrastructure" version="1.0.0.0" targetFramework="net48" />
-  <package id="MiniProfiler" version="4.2.1" targetFramework="net48" />
-  <package id="MiniProfiler.EF6" version="4.2.1" targetFramework="net48" />
-  <package id="MiniProfiler.Mvc5" version="4.2.1" targetFramework="net48" />
-  <package id="MiniProfiler.Shared" version="4.2.1" targetFramework="net48" />
```
Comment (Flemingjp) [inline src/Directory/Biobanks.Web/packages.config:63]: No need for the update here

File: src/Directory/Biobanks.Web/Views/PagesAdmin/CreateEditPreview.cshtml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    .top-padding {
+        margin-top: 50px;
+    }
+
+    .container {
+        padding-top: 25px;
+        padding-left: 25px;
+    }
+</style>
```
Comment (Flemingjp) [inline src/Directory/Biobanks.Web/Views/PagesAdmin/CreateEditPreview.cshtml:19]: Move to `Style.css`

File: src/Directory/Biobanks.Web/packages.config [hunk 3]
```diff
@@ -57,10 +57,10 @@
...
+  <package id="MiniProfiler" version="4.2.22" targetFramework="net48" />
+  <package id="MiniProfiler.EF6" version="4.2.22" targetFramework="net48" />
+  <package id="MiniProfiler.Mvc5" version="4.2.22" targetFramework="net48" />
+  <package id="MiniProfiler.Shared" version="4.2.22" targetFramework="net48" />
```
Comment (Flemingjp) [inline src/Directory/Biobanks.Web/packages.config:63]: No need for the update here

File: src/Directory/Biobanks.Web/Biobanks.Web.csproj [hunk 4]
```diff
@@ -161,16 +161,16 @@
       <HintPath>..\packages\Microsoft.Web.Infrastructure.1.0.0.0\lib\net40\Microsoft.Web.Infrastructure.dll</HintPath>
     </Reference>
     <Reference Include="MiniProfiler, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b44f9351044011a3, processorArchitecture=MSIL">
-      <HintPath>..\packages\MiniProfiler.4.2.1\lib\net461\MiniProfiler.dll</HintPath>
+      <HintPath>..\packages\MiniProfiler.4.2.22\lib\net461\MiniProfiler.dll</HintPath>
```
Comment (Flemingjp) [inline src/Directory/Biobanks.Web/Biobanks.Web.csproj:164]: No need for update here either

**Review by Flemingjp — CHANGES_REQUESTED** [review]
All looks good - someone just need to pickup changes to prevent the version updates

File: src/Directory/Biobanks.Web/Views/PagesAdmin/CreateEditPreview.cshtml (hunk 2 above)
Comment (Flemingjp) [reply src/Directory/Biobanks.Web/Views/PagesAdmin/CreateEditPreview.cshtml:17]: This is actually inline with all other Preview Views

**Review by beforan — APPROVED** [review]

**Review by blargly — APPROVED** [review]

**Pull request merged**